sqlglot/expressions.py

```diff
@@ -5640,6 +5640,10 @@ class DateAdd(Func, IntervalOp):
     arg_types = {"this": True, "expression": True, "unit": False}
 
 
+class DateBin(Func, IntervalOp):
+    arg_types = {"this": True, "expression": True, "unit": False}
+
```

Comment:
You need to make sure this doesn't break other dialects, e.g. clickhouse [supports](https://clickhouse.com/docs/en/sql-reference/functions/date-time-functions#tostartofinterval) `DATE_BIN` with four arguments, so you need to add a `"zone"` arg at the end of this dict.

Reply:
I'll do this as a follow-up, but make sure to test stuff like this for future PRs.